### ISSUE TITLE
Response to ask through actor selection times out

### ISSUE THREAD

**Issue by kosii** [akka/akka#16536]
The code below always times out:

```scala
implicit val t = Timeout(1 seconds)

val system = ActorSystem("system")
import system.dispatcher

system.actorOf(
  Props(
    new Actor {
      override def receive: Actor.Receive = {
        case r => context.actorSelection(sender.path) ! r
      }
    }
  )
) ? "hello" onComplete println
```

As far as I understood it's because the `VirtualPathContainer` does not reimplement the `receive` method. Is it a bug or it supposed to behave like this?
I did a quick fix and some unit tests. It apparently did not break existing tests, but I can't really evaluate the possible consequences and the correctness (for example I don't really know what `wildcardFanOut` does in an `ActorSelectionMessage`).

I implemented `receive` in `VirtualPathContainer` like this:
```scala
override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
  case ActorSelectionMessage(msg, elements, _) ⇒ {
    elements.foreach {
      case SelectChildName(name)       ⇒ getChild(name).tell(msg, sender)
      case SelectChildPattern(pattern) ⇒ // no idea
      case SelectParent                ⇒ getParent.tell(msg, sender)
    }
  }
}
```

**Comment by rkuhn** [comment]
Thanks for finding and raising this! Fixing this ticket could go either way:

* if it shall not work, then we need to reflect that in the docs (ScalaDoc and reference docs)
* if it shall work then we need to fix the code

Could you open a PR with your “quick fix” so we can have a look at what’s involved?

**Comment by kosii** [comment]
Sure, I'll do it this evening

**Comment by bantonsson** [comment]
Fixed by #16690

**Comment by kosii** [comment]
thanks for all the help, it was fun!
